### DISCUSSION TITLE
Is this a bug in Sequel::Postgres::Database#copy_table?

### DISCUSSION THREAD

**Discussion by yugoPanorama** [jeremyevans/sequel#2065] [General]
If I do this:

```ruby
<database>.copy_table(query, format: :csv) do |row| # query is a Sequel::Postgres::Dataset
  <method_that_runs_some_sequel_query>
  tmpfile << row
end
```
then I get a `PG::Error: no COPY in progress (Sequel::DatabaseDisconnectError)`.  If I remove the `method_that_runs_some_sequel_query`, all seems fine.

Not sure if this is expected.

**Comment by jeremyevans** [comment]
You are trying to issue a query while iterating over the results of a COPY on the same connection.  You should not do that.  You can do `results = Thread.new{method_that_runs_some_sequel_query}.join` to force a new connection (it's also possible by using Sequel's sharding support).

**Reply by yugoPanorama** [reply]
Thanks, @jeremyevans !
